spacy/pipeline/tagger.pyx

```diff
@@ -256,8 +256,14 @@ class Tagger(TrainablePipe):
         DOCS: https://nightly.spacy.io/api/tagger#get_loss
         """
         validate_examples(examples, "Tagger.get_loss")
-        loss_func = SequenceCategoricalCrossentropy(names=self.labels, normalize=False, missing_value="")
-        truths = [eg.get_aligned("TAG", as_string=True) for eg in examples]
+        loss_func = SequenceCategoricalCrossentropy(names=self.labels, normalize=False)
+        # Convert empty tag "" to missing value None so that both misaligned
+        # tokens and tokens with missing annotation have the default missing
+        # value None.
+        truths = []
+        for eg in examples:
+            eg_truths = [tag if tag is not "" else None for tag in eg.get_aligned("TAG", as_string=True)]
+            truths.append(eg_truths)
```

Comment:
I would specifically put `missing_value=None` when constructing `SequenceCategoricalCrossentropy`, for clarity but also to avoid future mismatches if the default would ever change in Thinc

Reply:
I can't imagine what value it would get changed to in the future and this seems like overkill to me? (We're not going to change the default ourselves without noticing that it breaks stuff?)

Reply:
Specifically defining a parameter that you're counting on to have a certain value, doesn't feel like overkill to me...

Reply:
But the value is coming from `Example` not `Tagger`, so it feels like it's just more places where this value is specified and things could go wrong?

Reply:
It's the same place though: your list comprehension specifically states `else None`, and it's this `None` value that you're counting on to be interpreted as `missing_value`.

Reply:
And yes there are too many additional dependencies on different types of "missing" values across the code base - hence my initial remark on making this more consistent internally ;-)

Reply:
I think we've pretty consistently used `None` as a missing value in places like this, but now we have to interpret missing values from `Doc` instead of `GoldParse`, which doesn't have the same nice clean `None` defaults. The `Scorer` has the default missing values as anything in `[None, 0, ""]`, but it's still tricky around the features that can't be unset (head, morph (in the python API), cats) or where there are some weird edge cases (sentence boundaries).

Reply:
Sure, but we're mixing up two conversations. Just from a software engineering perspective, this code decides to put `None` as missing value (instead of `""`) so I still say we should make it equally explicit to `SequenceCategoricalCrossentropy`, to ensure these two are always synced within these few lines of code. In my view it doesn't add a place where this could go wrong - it removes one. That's the reason why `SequenceCategoricalCrossentropy` has that parameter in the first place.

Reply:
Then I think it would be better to define something like `Example.MISSING_VALUE` and use that across the board instead of adding more literal `None` values that you have to keep in sync.

Reply:
One thing to remember is that we have to score missing and misaligned values differently, so there are some bits of code where we want to know the difference.

For instance, in the NER evaluation our scoring was slightly wrong for a long time, because we were ignoring gold-standard entities where we had tokenization misalignments, because they looked the same as missing values.

Reply:
The change here looks right to me

Reply:
Agreed that the change is right, my point was more about programming preferences. Marking as resolved in order not to hold up the PR!

Reply:
Ugh, right, I think I meant to write that on the top-level instead.